src/NuGet.Core/NuGet.Build.Tasks.Console/DependencyGraphSpecGenerator.cs

```diff
@@ -0,0 +1,893 @@
+
+            var packageSpec = new PackageSpec(restoreMetadataAndTargetFrameworkInformation.TargetFrameworkInfos)
+            {
+                FilePath = project.FullPath,
+                Name = restoreMetadataAndTargetFrameworkInformation.RestoreMetadata.ProjectName,
+                RestoreMetadata = restoreMetadataAndTargetFrameworkInformation.RestoreMetadata,
+                RuntimeGraph = new RuntimeGraph(
+                    MSBuildStringUtility.Split($"{project.GetProperty("RuntimeIdentifiers")};{project.GetProperty("RuntimeIdentifier")}")
+                        .Concat(projectsByTargetFramework.Values.SelectMany(i => MSBuildStringUtility.Split($"{i.GetProperty("RuntimeIdentifiers")};{i.GetProperty("RuntimeIdentifier")}")))
+                        .Distinct(StringComparer.Ordinal)
+                        .Select(rid => new RuntimeDescription(rid))
+                        .ToList(),
+                    MSBuildStringUtility.Split(project.GetProperty("RuntimeSupports"))
+                        .Distinct(StringComparer.Ordinal)
+                        .Select(s => new CompatibilityProfile(s))
+                        .ToList()
+                    ),
+                Version = GetProjectVersion(project)
+            };
+
+            return packageSpec;
+        }
+
+        /// <summary>
+        /// Gets the restore metadata and target framework information for the specified project.
+        /// </summary>
+        /// <param name="project">An <see cref="IMSBuildProject" /> representing the project.</param>
+        /// <param name="projectsByTargetFramework">A <see cref="IReadOnlyDictionary{NuGetFramework,IMSBuildProject}" /> containing the inner nodes by target framework.</param>
+        /// <param name="settings">The <see cref="ISettings" /> of the specified project.</param>
+        /// <returns>A <see cref="Tuple" /> containing the <see cref="ProjectRestoreMetadata" /> and <see cref="List{TargetFrameworkInformation}" /> for the specified project.</returns>
+        private (ProjectRestoreMetadata RestoreMetadata, List<TargetFrameworkInformation> TargetFrameworkInfos) GetProjectRestoreMetadataAndTargetFrameworkInformation(IMSBuildProject project, IReadOnlyDictionary<NuGetFramework, IMSBuildProject> projectsByTargetFramework, ISettings settings)
+        {
+            var projectName = GetProjectName(project);
+
+            var outputPath = GetRestoreOutputPath(project);
+
+            var targetFrameworkInfos = GetTargetFrameworkInfos(projectsByTargetFramework);
+
+            var projectStyleResult = BuildTasksUtility.GetProjectRestoreStyle(
+                restoreProjectStyle: project.GetProperty("RestoreProjectStyle"),
+                hasPackageReferenceItems: targetFrameworkInfos.Any(i => i.Dependencies.Any()),
+                projectJsonPath: project.GetProperty("_CurrentProjectJsonPath"),
```

Comment:
I don't think it will work at the moment.  But I pass this along for now.  I was too focused on getting this working at all to spend time on project.json.  In a future PR I can log an error or I can try and get it working.

Reply:
I don't necessarily think we want to care about project.json much, but we should get clarity on it. 
If you ask me, I want it gone by 6.0 :D 

https://github.com/NuGet/Home/issues/7199

Just worth tracking/calling out.